src/plugins/locale/index.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable dot-notation */
+import type { EsDayPlugin, Locale, UnitType } from 'esday'
+import * as C from '~/core/constant'
+import { prettyUnit, undefinedOr } from '~/core/utils'
+import en from '~/locales/en'
+
+const LocaleStore: Map<string, Locale> = new Map()
+
+let $localeGlobal = 'en'
+
+export function getLocale(name: string): Locale {
+  return LocaleStore.get(name) || en
+}
+
+export function registerLocale(locale: Locale, rename?: string): void {
+  LocaleStore.set(rename || locale.name, locale)
+}
+
+declare module 'esday' {
+/*   interface EsDay {
+    $locale: () => Locale
+  } */
+
+  interface EsDay {
+    locale: (localeName: string) => EsDay
+  }
+
+  interface EsDayFactory {
+    /**
+     * use locale as global
+     */
+    locale: (localeName: string) => EsDayFactory
+    /**
+     * register locale
+     */
+    registerLocale: (locale: Locale, rename?: string) => EsDayFactory
+  }
+}
+
+export const localePlugin: EsDayPlugin<{}> = (_, dayClass, dayFactory) => {
+  // @ts-expect-error $locale is private method
+  dayClass.prototype.$locale = function () {
+    // @ts-expect-error $l is private property
+    return getLocale(this['$l'] || 'en')
+  }
+
+  // add locale method
+  dayClass.prototype.locale = function (localeName: string) {
+    const inst = this.clone()
+    // @ts-expect-error $l is private property
+    inst['$l'] = localeName
+
+    return inst
+  }
+
+  // set $l in clone method
+  const oldClone = dayClass.prototype['clone']
+  dayClass.prototype['clone'] = function () {
+    const inst = oldClone.call(this)
+    // @ts-expect-error $l is private property
+    inst['$l'] = this['$l']
+    return inst
+  }
+
+  // set $l in parse method
```

Comment:
'parse' has a signature with a locale (see [dayjs documentation](https://day.js.org/docs/en/parse/string-format)).

Reply:
It seems a `CustomParseFormat` plugin feature.

Reply:
When all Locale dependent parsing / formatting functionalities (e.g. 'MMM') are removed from the core and transferred in a plugin (e.g. 'CustomParseFormat' and 'AdvancedFormat') then the current signature is ok.